crates/fluvio-smartengine/src/engine/wasmtime/instance.rs

```diff
@@ -189,6 +217,27 @@ impl SmartModuleInstanceContext {
         output.decode(&mut std::io::Cursor::new(bytes), self.version)?;
         Ok(output)
     }
+
+    pub(crate) fn metrics(&self) -> Arc<SmartModuleChainMetrics> {
+        self.metrics.clone()
+    }
+
+    /// convenience function for use with metrics_time_elapsed
+    pub(crate) fn metrics_time_start(&self) -> std::time::Instant {
+        std::time::Instant::now()
```

Comment:
is this code ran by wasm target? 

if yes, maybe we need:

```suggestion
#[cfg(not(target_arch = "wasm32"))]
std::time::Instant::now()
#[cfg(target_arch = "wasm32")]
web_time::Instant::now()
```

Reply:
I don't think we ever run this code anywhere but outside, but it's a good thing to verify - I'll do some more reading and verify a little better